R/er_los.R

```diff
@@ -68,10 +66,29 @@ er_los <- function(cohort, er) {
 
   ##### Prepare er data #####
   er[, `:=`(
-    triage_date_time = ymd_hm(triage_date_time),
-    left_er_date_time = ymd_hm(left_er_date_time)
+    triage_date_time = convert_dt(
+      triage_date_time, addtl_msg = ""
+    ),
```

Comment:
When running the function with `addtl_msg = NULL`, the generic default warning will be shown, which simply says `"Please carefully consider how to deal with missing/invalid date-time entries and perform any additional pre-processing prior to running the function (e.g., impute missing dates/timestamps etc.)"`. By specifying `addtl_msg = ""` here, I'm basically suppressing that generic warning, and instead, I'm showing a more specific warning within `er_los` that clarifies how these values are handled (see line ~80). 
I could in theory use that warning message on line ~80 as an input for `addtl_msg` but 1) it would be a bit repetitive in case the `er` input contains missing/invalid values in both `left_er_date_time` and `triage_date_time` (the message would be shown twice in that case) and 2) I wanted to mention the unique entries (i.e., `genc_ids` in this case) that are affected by missing/invalid entries in either variable. 
Does this clarify your question? Let me know if you think there is a better way of handling this.

Reply:
Thank you for the detail explanation! Very well-considered!